apps/rule-manager/client/src/ts/components/RuleForm.tsx

```diff
@@ -204,12 +207,24 @@ export const RuleForm = ({
 		onUpdate(ruleId);
 	};
 
-	const discardRuleChangesHandler = async () => {
+	const maybeDiscardRuleChangesHandler = () => {
 		if (!ruleId || ruleStatus !== 'live') {
 			return;
 		}
 
+		setIsRevertModalVisible(true);
+	};
+
+	const discardRuleChangesHandler = async () => {
+		const isDraftOrLive = ruleStatus === 'draft' || ruleStatus === 'live';
+		if (!ruleId || !isDraftOrLive) {
```

Comment:
Non-blocking: is this really a test for 'archived' status? If so, could we check for that explicitly?

Reply:
Good point, I suppose it actually needs to check if the status is 'live', as you can only revert a live rule. I'll change this, thanks for spotting.